### PR TITLE
test(connector-corda): fix flaky monitor transactions v4.8

### PR DESCRIPTION
**Pull request by outSH**
Fix race condition that caused user session removal during request processing,
that caused test to fail occasionaly.

Correlated fixes / improvements:
- Use paging when watching for vault changes.
This was necessary to perform long-running runs of a failing test.
  Without it, start monitor of a state with 200+ changes was failing instantly.
  I thought that paging was needed when there are 200+ changes in single observer push,
  but it seems that it breaks when total number of changes are that large,
  which makes it more critical to fix.
- To prevent deadlock of cleanup thread I introduced thread pool to corda kotlin server.
  Number of threads is configurable in environment variable.
  Updated readme to show this option.

Closes: #1995

Signed-off-by: Michal Bajer <michal.bajer@fujitsu.com>

### CODE REVIEW

**Comment by outSH** [comment]
@petermetz @izuru0 Please review.

Also, please upload image `ghcr.io/outsh/cactus-connector-corda-server:corda-dev-4` to hyperledger ghcr so I can update the reference in the test, and remove my name from cspell exclusion.

**Comment by petermetz** [comment]
> @petermetz @izuru0 Please review.

Also, please upload image `ghcr.io/outsh/cactus-connector-corda-server:corda-dev-4` to hyperledger ghcr so I can update the reference in the test, and remove my name from cspell exclusion.

@outSH Thank you for fixing the flake, much appreciated! Where is the diff for the `Dockerfile` changes?

**Review by petermetz — CHANGES_REQUESTED** [review]
> > Thank you for fixing the flake, much appreciated! Where is the diff for the `Dockerfile` changes?

There are no Dockerfile changes, I've changed files packed in `cactus-connector-corda-server` image that is used in the test.
 
@outSH Could you please make the changes in the Dockerfile so that the image you created is reproducible via the regular build?

**Review by petermetz — CHANGES_REQUESTED** [review]
> @petermetz Sorry, I don't understand. What changes in Dockerfile? Why the image wouldn't be reproducible via the regular build? 

@outSH No worries, I'm probably not explaining myself right (English isn't my first language so trouble sometimes ensues)

Let me try it again. 
Let's say we have image A (which is the image you get if you build the current dockerfile with the command from the readme).
Then let's say you have image B which is image A plus your changes ** in it.
The changes in image B are not done through the dockerfile and therefore if next week we change something in the dockerfile and publish image C, we would expect that image to be usable in place of image B and that everything would just work, but it wouldn't (from what I understand) because the changes you made in image B were done manually, not through the dockerfile and therefore when someone builds image C, they are not getting the changes that you made for B.

The key here is that technically image B is reproducible only if you do the regular build and then whatever manual steps you did** and what the expectation would be that image B (or any image that we have) is reproducible without manual extra steps apart from the build command itself.

** `I've changed files packed in cactus-connector-corda-server image that is used in the test.`

**Comment by outSH** [comment]
> The changes in image B are not done through the dockerfile and therefore if next week we change something in the dockerfile and publish image C

@petermetz 
The changes are replicable in dockerfile, but current image in hyperledger ghcr is older than one from this PR. I'm afraid I might be missing something important here - otherwise I'd just ask Izuru-san to upload this ;) - so I'll explain it in more detail:
- I have a functional test [monitor-transactions-v4.8.test.ts](https://github.com/hyperledger/cactus/pull/2021/files#diff-8f6f872134f8bc99de3d43129a25c4921928a9307314e8605a172498284f1709)
- This test used to pull `cactus-connector-corda-server` image from hyperledger ghcr, which contains kotlin connector (`packages/cactus-plugin-ledger-connector-corda/src/main-server`). This image is based on sources from couple commits ago.
- I've done some changes in `cactus-plugin-ledger-connector-corda/src/main-server` source files.
- If I run the test now, it would still use upstream ghcr  kotlin-server from couple commits ago, i.e. not including my changes.
- For this reason, I've built a new image, using the same, unmodified dockerfile, to have an image that represents kotlin-server state from this commit (i.e. including my changes).
- I've uploaded them to my ghcr, and updated the test, so that now the functional test uses the kotlin-server from this commit.
- I need to have this image uploaded to hyperledger ghcr, which will contain the source version of kotlin-server from this commit.
- I think normally this image would be updated with this PR sources after merge, but I want to have this earlier so we don't have to commit a test with reference to my GHCR.

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH Oh, okay, now I get it. Sorry, I thought you meant you were doing the changes with the `docker commit` command and that's why you wanted me to upload your custom image directly to ghcr.
If we are talking about me rebuilding the image myself and then uploading it, then it's all good of course.

Please update your image tag to this one and then it's LGTM: `ghcr.io/hyperledger/cactus-connector-corda-server:2022-05-26-0ff7407--pr-2021`

**Comment by outSH** [comment]
> @outSH Oh, okay, now I get it. Sorry, I thought you meant you were doing the changes with the `docker commit` command and that's why you wanted me to upload your custom image directly to ghcr. If we are talking about me rebuilding the image myself and then uploading it, then it's all good of course.

Oh, OK, now I see I'v asked you to upload my image from my `ghcr.io/outsh`, that was misleading, my bad. Good that we came to an understanding, eventually ;)

I've rebased, updated the image using in the test and fixed cspell ignore file. Should be good to go now.

**Review by petermetz — APPROVED** [review]
@outSH Haha, yeah we got it all together in the end! :-)  LGTM

**Review by izuru0 — APPROVED** [review]
LGTM

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

**Pull request merged**
